blockchain/fundamentals/ethereum-i/what-are-smart-contracts.md

```diff
@@ -12,20 +11,49 @@ type: normal
 category: must-know
 
 tags:
-
   - introduction
 
   - deep
-
 ---
+
 # What are Smart Contracts?
 
 ---
+
 ## Content
 
+**Smart contracts** are Ethereum's code files. They would be useless without Ethereum Virtual Machine. EVM is a runtime environment for executing compiled smart contracts: EVM does not run Solidity directly, but rather runs the *bytecode* obtained by compiling `.sol` files[1].
+
+Here's an example of a Solidity smart contract:
+```javascript
+pragma solidity ^0.4.23;
+
+contract MyAwesomeToken {
+  mapping (address => uint) public 
+    ownerToBalance;
+    
+  function setBalance(address owner,
+    uint newBalance) public {
+      require(msg.sender == owner);
+      ownerToBalance[owner] = newBalance;
+  }
 
+  function getBalance(address owner) public{
+    return ownerToBalance[owner];
+  }
+}
+```
+As you can see, it is similar to a Java code file. Once deployed, the contract would receive its own unique address. If someone were to send a transaction calling the contract's `setBalance` function using their *own* address and a random balance, the contract would create the mapping. 
```

Comment:
it might be worth saying something like:

The language being similar to Java, we can see ..... (explain what the code does)